az-hadoop-jobtype-plugin/src/main/java/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -433,52 +442,73 @@ public Void run() throws Exception {
    *
    * @return a Set. May be empty, but will never be null
    */
-  public static Set<String> findApplicationIdFromLog(String logFilePath, Logger log) {
-
-    File logFile = new File(logFilePath);
+  public static Set<String> findApplicationIdFromLog(final String logFilePath, final Logger log) {
+    // At least one job log file must be there.
+    final File logFile = new File(logFilePath);
 
     if (!logFile.exists()) {
       throw new IllegalArgumentException("the logFilePath does not exist: " + logFilePath);
     }
-    if (!logFile.isFile()) {
-      throw new IllegalArgumentException("the logFilePath specified  is not a valid file: "
-          + logFilePath);
-    }
-    if (!logFile.canRead()) {
-      throw new IllegalArgumentException(
-          "unable to read the logFilePath specified: " + logFilePath);
-    }
+
+    // Get the directory and file name
+    // The log files have names in this format,
+    // _job<job name specific string>.log
+    // A rolled over log file name appends to above format and adds an index in this format,
+    // _job<job name specific string>.log.<index>
+    // The index starts at 1 upto a configurable number.
+    final int lastSlash = logFilePath.lastIndexOf('/');
+    final String dirPath = logFilePath.substring(0, lastSlash);
+    final String logFileName = logFilePath.substring(lastSlash + 1);
+
+    // Fetch all the log files for this job which start with
+    // _job<job name specific string>.log
+    final File[] logFiles =
+        new File(dirPath).listFiles((dir, name) -> name.startsWith(logFileName));
 
     BufferedReader br = null;
-    Set<String> applicationIds = new HashSet<String>();
+    final Set<String> applicationIds = new HashSet<>();
 
     try {
-//      br = new BufferedReader(new FileReader(logFile));
-      br = new BufferedReader(new InputStreamReader(
-          new FileInputStream(logFile), StandardCharsets.UTF_8));
-      String line;
-
-      // finds all the application IDs
-      while ((line = br.readLine()) != null) {
-        String[] inputs = line.split("\\s");
-        if (inputs != null) {
-          for (String input : inputs) {
-            Matcher m = APPLICATION_ID_PATTERN.matcher(input);
+      // There can be more than one log file. Go through each one of them.
+      for (final File curLogFile : logFiles) {
+        // Start with sanity checks
+        if (!curLogFile.exists()) {
```

Comment:
When are rolled over files deleted? Is it possible that one gets deleted right after it's being listed but before we try to read it? In that case shouldn't we try reading the next file available instead of failing fast?